objects/src/transaction/transaction_id.rs

```diff
@@ -0,0 +1,148 @@
+use super::{
+    Digest, Felt, Hasher, ProvenTransaction, TransactionResult, Vec, Word, WORD_SIZE, ZERO,
+};
+use crate::utils::serde::{
+    ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+};
+
+// TRANSACTION ID
+// ================================================================================================
+
+/// A unique identifier of a transaction.
+///
+/// Transaction ID is computed as:
+///
+/// hash(init_account_hash, final_account_hash, input_notes_hash, output_notes_hash)
+///
+/// This achieves the following properties:
+/// - If two transactions have the same ID, the transactions must be identical.
+/// - If two transaction have different IDs, the transactions must be distinct.
```

Comment:
Maybe you can merge those two points to: 

Transactions must be identical if and only if they have the same ID

Reply:
Agreed! This is better. Changed.